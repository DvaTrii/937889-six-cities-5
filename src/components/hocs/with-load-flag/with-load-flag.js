import React from "react";
import {Spinner} from "../../spinner/spinner";

export const withLoadFlag = (Component) => (props) => {
  const {isLoaded} = props;
  return isLoaded ? <Component {...props} /> : <Spinner />;
};
