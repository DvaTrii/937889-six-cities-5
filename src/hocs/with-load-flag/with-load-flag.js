import React from "react";
import {Spinner} from "../../components/spinner/spinner";

export const withLoadFlag = (Component) => (props) => {
  const {isLoaded} = props;
  return isLoaded ? <Component {...props} /> : <Spinner />;
};
