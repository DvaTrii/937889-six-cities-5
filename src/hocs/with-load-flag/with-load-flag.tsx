import * as React from "react";
import {Spinner} from "../../components/spinner/spinner";

export const withLoadFlag = (Component: typeof React.Component) => (props) => {
  const {isLoaded} = props;
  return isLoaded ? <Component {...props} /> : <Spinner />;
};
