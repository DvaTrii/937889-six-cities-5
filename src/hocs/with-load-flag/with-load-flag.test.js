import React from "react";
import renderer from "react-test-renderer";
import {withLoadFlag} from "./with-load-flag.tsx";

const MockComponent = () => {

  return (
    <React.Fragment>
    </React.Fragment>
  );
};

const MockComponentWrapped = withLoadFlag(MockComponent);

it(`withLoadFlag is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped isLoad={true}/>
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
