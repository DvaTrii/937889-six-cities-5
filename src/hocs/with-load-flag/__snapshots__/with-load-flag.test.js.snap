// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoadFlag is rendered correctly 1`] = `
<div
  className="lds-ring"
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
